src/controllers/__tests__/translation.test.ts

```diff
@@ -0,0 +1,62 @@
+import axios from 'axios';
+import {
+  requestFixture,
+  responseFixture,
+  nextFunctionFixture,
+} from '../../../__tests__/shared/fixtures';
+import { getTranslation } from '../translation';
+
+describe('translation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+  it('calls Nkọwa okwu ML translation server', async () => {
+    const req = requestFixture({
+      body: { igbo: 'aka' },
+      headers: {
+        'Content-Type': 'application/json',
+        'X-API-Key': 'main_key',
+      },
+    });
+    const res = responseFixture();
+    const next = nextFunctionFixture();
+    jest.spyOn(axios, 'request').mockResolvedValue({
+      data: { igbo: 'aka' },
+    });
+    await getTranslation(req, res, next);
+    expect(res.send).toHaveBeenCalled();
+  });
+
+  it('throws validation error when input is too long', async () => {
+    const req = requestFixture({
+      body: { igbo: 'aka'.repeat(100) },
+      headers: {
+        'Content-Type': 'application/json',
+        'X-API-Key': 'main_key',
+      },
+    });
+    const res = responseFixture();
+    const next = nextFunctionFixture();
+    jest.spyOn(axios, 'request').mockResolvedValue({
+      data: { igbo: 'aka'.repeat(100) },
+    });
+    await getTranslation(req, res, next);
+    expect(next).toHaveBeenCalled();
```

Comment:
is it possible to check the error message here too so we know which error was thrown?

Reply:
now doing that check!